Packs/AWS-Enrichment-Remediation/Scripts/AWSRecreateSG/AWSRecreateSG.py

```diff
@@ -0,0 +1,287 @@
+            else:
+                raise ValueError('Error on adding security group ingress rules to new security group')
+    # Check if there was a rule for `all traffic` (added by default), but break if it is the only egress rule.
+    match_all_trafic = False
+    for egress in info['IpPermissionsEgress']:
+        if egress["IpProtocol"] == '-1':
+            if len(info['IpPermissionsEgress']) == 1:
+                break
+            match_all_trafic = True
+        e_format = str([egress]).replace("'", "\"")
+        res = demisto.executeCommand("aws-ec2-authorize-security-group-egress-rule",
+                                     {"groupId": new_id, "IpPermissionsFull": e_format})
+        # Don't error if the message is that the rule already exists.
+        if isError(res):
+            if "already exists" in res[0]['Contents']:
+                pass
+            else:
+                raise ValueError('Error on adding security group egress rules to new security group')
+    # If `all traffic` rule before, remove the default one.
+    if match_all_trafic is True:
+        all_traffic_rule = """[{"IpProtocol": "-1", "IpRanges": [{"CidrIp": "0.0.0.0/0"}], "Ipv6Ranges": [],""" + \
+                           """"PrefixListIds": [], "UserIdGroupPairs": []}]"""
+        res = demisto.executeCommand("aws-ec2-revoke-security-group-egress-rule",
+                                     {"groupId": new_id, "IpPermissionsFull": all_traffic_rule})
+        if isError(res):
+            raise ValueError('Error on removing egress `allow all` rule on new security group')
+    return {'new-sg': new_id}
+
+
+def replace_sgs(replace_list: List, int_sg_mapping: Dict):
+    """
+    Replace the actual SGs on the interface
+
+    Args:
+        replace_list (List): list of SGs to be replaced
+        int_sg_mapping: interface to security group mapping
+
+    Returns:
+        none
+    """
+    for entry in replace_list:
+        int_sg_mapping[entry['int']].remove(entry['old-sg'])
+        int_sg_mapping[entry['int']].append(entry['new-sg'])
+        formatted_list = ','.join(int_sg_mapping[entry['int']])
+        res = demisto.executeCommand("aws-ec2-modify-network-interface-attribute",
+                                     {"networkInterfaceId": entry['int'], "groups": formatted_list})
+        if isError(res):
+            raise ValueError('Error on replacing security group(s) on network interface')
+
+
+def determine_excessive_access(int_sg_mapping: Dict, port: int, protocol: str) -> List:
+    """
+    Pulls info on each SG and then calls sg_fix() to actually create the new SGs
+
+    Args:
+        int_sg_mapping (Dict): interface to security group mapping
+        port (int): TCP/UDP Port to be restricted
+        protocol (str): Protocol of the port to be restricted
+
+    Returns:
+        List: list of SGs to be replaced
+    """
+    replace_list = []
+    for mapping in int_sg_mapping.keys():
+        for sg in int_sg_mapping[mapping]:
```

Comment:
```suggestion
    for mapping in int_sg_mapping.values():
        for sg in mapping:
```
This change is optional, but this way the code is easier to read :)

Reply:
I made this change and unit tests were breaking.  Will keep as-is